src/suggestion-box.ts

```diff
@@ -0,0 +1,409 @@
+import { LitElement, html, css } from "lit";
+import { customElement, property, state } from "lit/decorators.js";
+import { queryAssignedElements } from "lit/decorators.js";
+
+@customElement("aalam-sgn-box")
+export class SuggestionBox extends LitElement {
+    static override styles = css`
+        :host {
+            position: relative;
+        }
+    `;
+
+    @property({ type: Array<Object> })
+    list: Array<{ [key: string]: any }> = [];
+
+    @property()
+    listkey: string = "";
+
+    @queryAssignedElements({ slot: "sgn-item-template" })
+    template_slots: Array<HTMLElement>;
+
+    @queryAssignedElements({ slot: "__private-item" })
+    private_items: Array<HTMLElement>;
+
+    @property()
+    size: number = 0;
```

Comment:
There is no need for this. You seem to be using it in two places and in both the cases you are just setting to this variable and reading from it.